### PR TITLE
add addLiquidity variants

### PR DESCRIPTION
**Pull request by jrhea**
# Description

This PR adds the delta and state variants to addLiquidity

# Review Checklists

Please check each item **before approving** the pull request. While going
through the checklist, it is recommended to leave comments on items that are
referenced in the checklist to make sure that they are reviewed.

- [ ] **Testing**
    - [x] Are there new or updated unit or integration tests?
    - [ ] Do the tests cover the happy paths?
    - [ ] Do the tests cover the unhappy paths?
    - [ ] Are there an adequate number of fuzz tests to ensure that we are
          covering the full input space?
    - [ ] If matching Solidity behavior, are there differential fuzz tests that
          ensure that Rust matches Solidity?

### CODE REVIEW

File: crates/hyperdrive-math/src/lp.rs [hunk 1]
```diff
@@ -74,6 +58,64 @@ impl State {
...
+
+    pub fn calculate_pool_deltas_after_add_liquidity(
+        &self,
+        current_block_timestamp: U256,
+        contribution: FixedPoint,
+        min_lp_share_price: FixedPoint,
+        min_apr: FixedPoint,
+        max_apr: FixedPoint,
+        as_base: bool,
+    ) -> Result<(FixedPoint, I256, FixedPoint), &'static str> {
```
Comment (sentilesdal) [inline crates/hyperdrive-math/src/lp.rs:93]: oops, you didn't do this but make this:
```rust
    ) -> Result<(FixedPoint, I256, FixedPoint)> {
```
and in `calculate_update_liquidity` too.
then in calculate_update_liquidity wrap the `Err("")` with `Err(eyre!(""))`

File: crates/hyperdrive-math/src/lp.rs [hunk 2]
```diff
@@ -74,6 +58,64 @@ impl State {
...
+        let (share_reserves, share_adjustment, bond_reserves) = self
+            .calculate_update_liquidity(
+                self.share_reserves(),
+                self.share_adjustment(),
+                self.bond_reserves(),
+                self.minimum_share_reserves(),
+                I256::try_from(0).unwrap(),
+            )
+            .unwrap();
```
Comment (sentilesdal) [inline crates/hyperdrive-math/src/lp.rs:102]: use `?` instead of `unwrap()` to gracefully handle errors.

File: crates/hyperdrive-math/src/lp.rs [hunk 3]
```diff
@@ -74,6 +58,64 @@ impl State {
...
+        let (new_share_reserves, new_share_adjustment, new_bond_reserves) = self
+            .calculate_update_liquidity(
+                self.share_reserves(),
+                self.share_adjustment(),
+                self.bond_reserves(),
+                self.minimum_share_reserves(),
+                I256::try_from(contribution).unwrap(),
+            )
+            .unwrap();
```
Comment (sentilesdal) [inline crates/hyperdrive-math/src/lp.rs:111]: use `?`.

File: crates/hyperdrive-math/src/lp.rs (hunk 1 above)
Comment (jrhea) [reply crates/hyperdrive-math/src/lp.rs]: okay done. can you take a look again?

**Review by sentilesdal — APPROVED** [review]
lgtm!

**Review by dpaiton — APPROVED** [review]

**Pull request merged**